### ISSUE TITLE
Import Error

### ISSUE THREAD

**Issue by josephdunbar** [rusty1s/pytorch_geometric#916]
#301  📚 Installation

## Environment

* OS: Windows 10
* Python version: 3.7.4
* PyTorch version: 1.4.0
* CUDA/cuDNN version: 10.1
* GCC version: 5.3.0
* How did you try to install PyTorch Geometric and its extensions (pip, source): pip
* Any other relevant information: Anaconda3 environment

## Checklist

- [ X ] I followed the installation guide.
- [ X ] I cannot find my error message in the FAQ.
- [ ] I set up CUDA correctly and can compile CUDA code via `nvcc`.
- [ ] I have cloned the repository and tried a manual installation from source.
- [ ] I do have multiple CUDA versions on my machine.
- [ ] I checked if the official extension example runs on my machine.
- [ ] The offical extension example runs on my machine.

## Error Shown when I try to import Data from torch_geometric.data in a jupyter notebook

ImportError: DLL load failed: The specified module could not be found.

The traceback for this error is shown below:

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-b336892afbe1> in <module>()
----> 1 from torch_geometric.data import Data

C:\Users\Owner\Anaconda3\lib\site-packages\torch_geometric\__init__.py in <module>()
      1 from .debug import is_debug_enabled, debug, set_debug
----> 2 import torch_geometric.nn
      3 import torch_geometric.data
      4 import torch_geometric.datasets
      5 import torch_geometric.transforms

C:\Users\Owner\Anaconda3\lib\site-packages\torch_geometric\nn\__init__.py in <module>()
      1 from .meta import MetaLayer
----> 2 from .data_parallel import DataParallel
      3 from .reshape import Reshape
      4 from .conv import *  # noqa
      5 from .glob import *  # noqa

C:\Users\Owner\Anaconda3\lib\site-packages\torch_geometric\nn\data_parallel.py in <module>()
      3 
      4 import torch
----> 5 from torch_geometric.data import Batch
      6 
      7 

C:\Users\Owner\Anaconda3\lib\site-packages\torch_geometric\data\__init__.py in <module>()
----> 1 from .data import Data
      2 from .batch import Batch
      3 from .dataset import Dataset
      4 from .in_memory_dataset import InMemoryDataset
      5 from .dataloader import DataLoader, DataListLoader, DenseDataLoader

C:\Users\Owner\Anaconda3\lib\site-packages\torch_geometric\data\data.py in <module>()
      5 import torch
      6 import torch_geometric
----> 7 from torch_sparse import coalesce
      8 from torch_geometric.utils import (contains_isolated_nodes,
      9                                    contains_self_loops, is_undirected)

C:\Users\Owner\Anaconda3\lib\site-packages\torch_sparse\__init__.py in <module>()
      1 from .convert import to_torch_sparse, from_torch_sparse, to_scipy, from_scipy
----> 2 from .coalesce import coalesce
      3 from .transpose import transpose
      4 from .eye import eye
      5 from .spmm import spmm

C:\Users\Owner\Anaconda3\lib\site-packages\torch_sparse\coalesce.py in <module>()
      1 import torch
----> 2 import torch_scatter
      3 
      4 from .utils.unique import unique
      5 

C:\Users\Owner\Anaconda3\lib\site-packages\torch_scatter\__init__.py in <module>()
      1 from .add import scatter_add
      2 from .sub import scatter_sub
----> 3 from .mul import scatter_mul
      4 from .div import scatter_div
      5 from .mean import scatter_mean

C:\Users\Owner\Anaconda3\lib\site-packages\torch_scatter\mul.py in <module>()
      1 from torch.autograd import Function
      2 
----> 3 from torch_scatter.utils.ext import get_func
      4 from torch_scatter.utils.gen import gen
      5 

C:\Users\Owner\Anaconda3\lib\site-packages\torch_scatter\utils\ext.py in <module>()
      1 import torch
----> 2 import torch_scatter.scatter_cpu
      3 
      4 if torch.cuda.is_available():
      5     import torch_scatter.scatter_cuda

## Additional context

I had many warnings about "dll" and "host" when installing torch-scatter, torch-sparse, torch-cluster, and torch-spline-conv, but it seemed to install ok. Apparently not. Please help, as I don't know how to fix this.

For Example:

C:/Users/Owner/Anaconda3/lib/site-packages/torch/include\ATen/Context.h(132): warning: field of class type without a DLL interface used in a class with a DLL interface

And:

C:/Users/Owner/Anaconda3/lib/site-packages/torch/include\ATen/core/ivalue_inl.h(313): warning: base class dllexport/dllimport specification differs from that of the derived class

And:

C:/Users/Owner/Anaconda3/lib/site-packages/torch/include\c10/util/BFloat16.h(48): warning: calling a __host__ function from a __host__ __device__ function is not allowed

And:

C:/Users/Owner/Anaconda3/lib/site-packages/torch/include\c10/core/TensorTypeSet.h(44): warning: integer conversion resulted in a change of sign

And:

C:/Users/Owner/Anaconda3/lib/site-packages/torch/include\c10/util/typeid.h(460): warning: dllexport/dllimport conflict with "caffe2::TypeMeta::_typeMetaDataInstance [with T=caffe2::detail::_Uninitialized]"
  (456): here; dllexport assumed

And:

C:\Users\Owner\Anaconda3\lib\site-packages\torch\include\torch/csrc/autograd/custom_function.h(113): warning C4251: 'torch::autograd::AutogradContext::grad_fn_': class 'std::weak_ptr<torch::autograd::Node>' needs to have dll-interface to be used by clients of struct 'torch::autograd::AutogradContext'

Hope these help!

**Comment by rusty1s** [comment]
This seems compiler related. Which version of cl are you using? We have many resolved issues and advices regarding Windows installation in the issues, so maybe this might be also helpful to you.

**Comment by josephdunbar** [comment]
Thank you for the prompt reply. After trying several of the Windows solutions on these forums, and having them not work, I decided to setup up a dual-boot with Windows 10 and Ubuntu. After disabling Wayland and Nouveau, downloading the latest driver from NVIDIA, (compatible with CUDA 10.2), I was able to get this to work. Note that this did not require that my CUDA and Cuda toolkit for pytorch match, as the command...

conda install pytorch torchvision cudatoolkit=10.2 -c pytorch

...downloaded cudatoolkit 9.0. I used pip for my system python and pytorch ("pip install torch torchvision"), followed the rest of the PyG installation instructions, and I now have a working pytorch geometric that works within the jupyter notebook environment.

I spent an exorbitant amount of time trying to download CUDA 10.1 and match it with cudatoolkit 10.1, only to have it be incompatible with the driver currently being offered by NVIDIA. I think that the pytorch geometric installation instructions should reflect the fact that these need not match in order to have a working installation of PyG. I am now closing the issue, and thank you for your time.
